docs/source/WritingTests/WritingSimpleTests.rst

```diff
@@ -167,6 +162,42 @@ to pick up a living guest, connect to it via ssh, and return its uptime.
            logging.info("Guest uptime result is: %s", uptime)
            session.close()
 
+#. Note that all failures that might happen here are implicitly handled
+   by the methods called. If a test went from its beginning to its end
+   without unhandled exceptions, avocado assumes the test automatically
+   as **PASS**, *no need to mark a test as explicitly passed*. If you
+   have explicit points of failure, for more complex tests, you might
+   want to mark it explicitly. ``test.skip`` make the test **SKIP**,
+   ``test.error`` make the test **Error**, and ``test.fail`` make the
+   test **Fail**. Note, people should never import ``exceptions``
+   from ``avocado.core`` to raise them in test case, see `avocado doc <http://avocado-framework.readthedocs.io/en/latest/api/core/avocado.core.html>`_
+   for more details. *BTW, check the uptime makes no sense, but let's continue
+   this example for test status explanation*:
+
+   .. code-block:: python
+      :emphasize-lines: 16,18,20
+
+       def run(test, params, env):
+           """
+           Docstring describing uptime.
+           """
+           vm = env.get_vm(params["main_vm"])
+           vm.verify_alive()
+           timeout = float(params.get("login_timeout", 240))
+           session = vm.wait_for_login(timeout=timeout)
+           uptime = session.cmd('uptime')
+           logging.info("Guest uptime result is: %s", uptime)
+           session.close()
+           expected_skip_msg = '0 min'
+           expected_err_msg = '1 day'
+           expected_fail_msg = '10 days'
+           if expected_skip_msg in uptime:
+               test.skip('Skip message')
+           if expected_err_msg in uptime:
+               test.error('Error message')
+           if expected_fail_msg in uptime:
+               test.fail('Fail message')
```

Comment:
The use of `test` is really something we did not anticipate, but as a public API it should be supported IMO.  And @ldoktor is indeed right about the `skip` status, it will affect this use case in the future.

As another notice, some exceptions have been moved to the test API recently (public and supported): https://github.com/avocado-framework/avocado/pull/1875